zerver/lib/topic_link_util.py

```diff
@@ -0,0 +1,79 @@
+import re
```

Comment:
We should add comments at the top of this and the JS implementation noting that we aim to keep them in sync.